src/templates/post.js

```diff
@@ -113,6 +119,26 @@ const BlogPost = ({ data }) => {
     }
   };
 
+  const getActiveTitle = () => {
+    if (typeof window !== 'undefined' && typeof window.document !== "undefined") {
+      const postContainer = document.getElementById('postContainer');
+      if (postContainer) {
+        const elementList = Array.from(postContainer.childNodes);
+        const titles = findTitles(elementList);
```

Comment:
@lauta2912 o @gpicart I have a question, for you, Is it better to make a hook that returns the list of titles and use it in the two functions I have or no?

Reply:
First of all, is it extremely needed to do this by manipulating the real DOM?
Ate least, you can avoid calling document.getElementById by using React refs